src/engine/strat_engine/backstore/devices.rs

```diff
@@ -823,16 +823,27 @@ mod tests {
     // Verify that if the last device in a list of devices to intialize
     // can not be intialized, all the devices previously initialized are
     // properly cleaned up.
-    fn test_failure_cleanup(paths: &[&Path]) {
-        assert!(paths.len() > 1);
+    fn test_failure_cleanup(paths: &[&Path], key_desc: Option<&str>) -> Result<(), Box<dyn Error>> {
+        if paths.len() <= 1 {
+            return Err(Box::new(StratisError::Error(
+                "Test requires more than one device".to_string(),
+            )));
+        }
 
-        let mut devices: Vec<DeviceInfo> = process_devices(paths)
-            .unwrap()
+        let mut devices: Vec<DeviceInfo> = process_devices(paths)?
             .into_iter()
             .map(|(info, _)| info)
             .collect::<Vec<DeviceInfo>>();
 
-        let old_info = devices.pop().unwrap();
+        if devices.len() != paths.len() {
+            return Err(Box::new(StratisError::Error(
+                "Not all devices were successfully processed".to_string(),
+            )));
+        }
+
+        let old_info = devices
+            .pop()
+            .ok_or_else(|| StratisError::Error("No devices were processed".to_string()))?;
```

Comment:
Please put an expect() here. There has to be one, or the method would have returned. Something like ```.expect(devices.len() is at least one")```.

Reply:
The idea here is to remove all panicking methods wherever possible. If it panics, it will probably leave the keyring in an unusable state based on the tests that I've done. Expired keys can be a little bit hard to clean up. We want it to return on any error instead of panicking so that the key cleanup can occur.

Reply:
If you're saying we've already done adequate checks to assert that this will always be the case, that makes a little bit more sense.

Reply:
> If you're saying we've already done adequate checks to assert that this will always be the case, that makes a little bit more sense.

Yes. it is a logical necessity. If pop() doesn't yield a value, then we have to conclude that the standard library is unreliable in its len() methods, and give up the whole enterprise.

Reply:
Resolved.